pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -2190,6 +2190,14 @@ impl RemoteTimelineClient {
                     upload_queue.clean.1 = Some(task.task_id);
 
                     let lsn = upload_queue.clean.0.metadata.disk_consistent_lsn();
+                    crate::metrics::PROJECTED_REMOTE_CONSISTENT_LSN
+                        .get_metric_with_label_values(&[
+                            &self.tenant_shard_id.tenant_id.to_string(),
+                            &format!("{}", self.tenant_shard_id.shard_slug()),
+                            &self.timeline_id.to_string(),
+                        ])
+                        .unwrap()
+                        .set(lsn.0 as i64);
```

Comment:
Please integrate into `self.metrics`. Otherwise we won't clean up.

Reply:
The metric itself is de-registered in `TimelineMetrics::shutdown`. The metrics shut down last and by that point the remote timeline client should have exited. I can thread the gauge through to here if you insist though.

Reply:
Yeah, I would prefer we don't get advanterous with the metrics lifecycles.

This should be part of `RemoteTimelineClientMetrics` and be hooked into its lifecycle machinery (`impl Drop for RemoteTimelineClientMetrics`).

Reply:
Don't forget to remove it from

https://github.com/neondatabase/neon/blob/5b376379e5c571ce686b8358670915f3396eff35/pageserver/src/metrics.rs#L2670-L2674

Reply:
https://github.com/neondatabase/neon/pull/10005/commits/4f74ea9a9bd5c050ca37471dedf01376a57dc0b4